src/sql/azdata.d.ts

```diff
@@ -46,7 +46,7 @@ declare module 'azdata' {
 		 * @param providerId The ID that the provider was registered with
```

Comment:
Keep in mind most of these are considered "breaking changes" - not sure what we want to do about that.

Reply:
They are "compiler" breaks, but they are functionally more correct

Reply:
If a CI system can't rebuild the same code, it's a breaking change. Even though it has no impact on the running code.

Reply:
Yes and no, in this instance an extension would have to manually upgrade their typings, and then it would break. Its not a matter of we check something in and then on no action of their own, their CI/extension breaks. This is refining our current typings to be more accurate. We have done this in the past (although not on such a large scale). VSCode also has does this on occasion; so there is plenty of precedent.

Reply:
We're not providing an API as a service, so in our case breaking update means that once the user updates their API typings, stuff will start failing for them (assuming they had strict compilation rules on).

So... yes, they'd have to update their typings. Which is what breaking API changes mean. Going from some version of an SDK to a higher version of the SDK and existing behavior breaking is by definition a breaking change.

I'm not saying we should avoid all breaking changes - but more so this needs to be communicated to the developers of third party extensions that they may face compilation issues when they update to the new version of typings.

Reply:
Again, yes and no, 
> Going from some version of an SDK to a higher version of the SDK and existing behavior breaking is by definition a breaking change.

This is not true, there is no behavior changes here. Installing the same vsix from before and after these changes will see no change. 

By using the typings we have provided, the extension authors are implicitly saying they want to know that the are functionality staying in line with what we are functionally doing. So updating our typing to be more functionally correct is a all around benefit to extension authors.

There has to be a difference between typing changes and behavioral changes. Behavioral breaks should be avoided at all costs except on a major version bump. Typing changes that lead to more accurate code should be updated whenever possible.

If extension authors didn't want to make sure they are interacting with the product in the right way should just declare a global module
``` typescript
declare module 'azdata' {
       {[key: string]}: any;
}
```

Reply:
> There has to be a difference between typing changes and behavioral changes. 

I never said there isn't. Behavior isn't the only thing that defines a contract of an API. The typings themselves are a contract.

What I'm suggesting is that on the release notes (and earlier through email with the few partners that have provided us points of contact), we let them know that this change is incoming but they shouldn't see any behavior differences. 

> If extension authors didn't want to make sure they are interacting with the product in the right way should just declare a global module

This is not a good statement for people who have put trust in our APIs and and expecting the compile step of their applications to not just crash when they do an update without any sort of notice.

Our azdata types are a part of our API. They don't have any functional value, but they do have developer and compiling value. What I'm suggesting we do here is simply to let them know that this PR is going to be merged for the September release and what they should expect if they have strict compiles on and how to mitigate any problems that arise. I am **not** against making updates to types to correct them.